backend/src/connectors/fileScanning/modelScan.ts

```diff
@@ -0,0 +1,206 @@
+import fetch, { Response } from 'node-fetch'
+import { Readable } from 'stream'
+
+import { getObjectStream } from '../../clients/s3.js'
+import { FileInterfaceDoc, ScanState } from '../../models/File.js'
+import log from '../../services/log.js'
+import config from '../../utils/config.js'
+import { BadReq, ConfigurationError, InternalError } from '../../utils/error.js'
+import { BaseFileScanningConnector, FileScanResult } from './Base.js'
+
+let av: NodeModelScanAPI
+export const modelScanToolName = 'ModelScan'
+
+interface ModelScanOptions {
+  host: string
+  port: number
+  protocol: string
+}
+
+type ModelScanInfoResponse = {
+  apiName: string
+  apiVersion: string
+  scannerName: string
+  modelscanVersion: string
+}
+
+type ModelScanResponse = {
+  summary: {
+    total_issues: number
+    total_issues_by_severity: {
+      LOW: number
+      MEDIUM: number
+      HIGH: number
+      CRITICAL: number
+    }
+    input_path: string
+    absolute_path: string
+    modelscan_version: string
+    timestamp: string
+    scanned: {
+      total_scanned: number
+      scanned_files: string[]
+    }
+    skipped: {
+      total_skipped: number
+      skipped_files: string[]
+    }
+  }
+  issues: [
+    {
+      description: string
+      operator: string
+      module: string
+      source: string
+      scanner: string
+      severity: string
+    },
+  ]
+  // TODO: currently unknown what this might look like
+  errors: object[]
+}
+
+class NodeModelScanAPI {
```

Comment:
I'm not a huge fan of having multiple classes in one file and I do think we necessarily need to use a class here. Rather than using a class I would prefer we created a modelScan client in https://github.com/gchq/Bailo/tree/main/backend/src/clients . Most other clients we have follow a pretty similar pattern to this in wanting to define a simple set of methods that make http requests and return the data.